core/modules/image/image.field.inc

```diff
@@ -542,9 +556,54 @@ function image_field_widget_process($element, &$form_state, $form) {
     '#access' => (bool) $item['fid'] && $settings['title_field'],
   );
 
+  // Pass alt and title required status to JS in case other modules need access.
+  $element['#attached']['js'][] = array(
+    'type' => 'setting',
+    'data' => array(
+      'image' => array(
+        'altRequired' => $instance['settings']['alt_field_required'],
+      ),
+    ),
+  );
```

Comment:
This looks entirely optional (I'm guessing matching Drupal), but I think we could accomplish this better by putting a data attribute on the alt field itself rather than making a page-level JS setting. Page-level settings can get finicky with modals and AJAX requests.

So I'd suggest something like this:
```
<input type="text" id="edit-field-image-und-0-alt" name="field_image[und][0][alt]" value="" size="60" maxlength="512" class="form-text" data-alt-required="true">
```

Which can be achieved with `'#attributes' => array('data-alt-required' => 'true')`

Reply:
Fixed in the alternative PR with this commit: [`5fc5e13` (#4422)](https://github.com/backdrop/backdrop/pull/4422/commits/5fc5e139ef98bbf53a90033850367b3b52fe8b55)